backend/src/main/java/de/iav/backend/service/TicketService.java

```diff
@@ -45,7 +54,24 @@ public Ticket updateTicketById(String ticketId, TicketRequestDTO ticketToUpdate)
             );
             return ticketRepository.save(updatedTicket);
         }
-        throw new NoSuchElementException("Element with " + ticketId + " not found!");
+        throw new NoSuchElementException(EXCEPTION_MESSAGE_PART1 + ticketId + EXCEPTION_MESSAGE_PART2);
+    }
+
+    public Ticket updateTicketStatusById(String ticketId, TicketRequestDTO ticketToUpdate) {
+        Optional<Ticket> existingTicket = ticketRepository.findById(ticketId);
+        if (existingTicket.isPresent()) {
+            Ticket updatedTicket = new Ticket(
+                    existingTicket.get().id(),
+                    existingTicket.get().subject(),
+                    existingTicket.get().priority(),
+                    ticketToUpdate.status(),
+                    existingTicket.get().text(),
+                    existingTicket.get().creatorId(),
+                    existingTicket.get().creationDate()
+            );
+            return ticketRepository.save(updatedTicket);
+        }
+        throw new NoSuchElementException(EXCEPTION_MESSAGE_PART1 + ticketId + EXCEPTION_MESSAGE_PART2);
```

Comment:
Nur das du es mal gesehen hast: Es gibt ne abgefahrene Möglichkeiten den "Syntax Sugar" der Optionals zu verwenden und mit `orElse` & co erlauben auf das "händische" `if` zu verzichten.

Ob das besser ist, steht zur Frage, du brauchst es auch nicht übernehmen, aber so wäre eine mögliche Alterantive:


```java
return ticketRepository.findById(ticketId)
        .map(existingTicket -> new Ticket(
                existingTicket.id(),
                ticketToUpdate.subject(),
                ticketToUpdate.priority(),
                ticketToUpdate.status(),
                ticketToUpdate.text(),
                ticketToUpdate.creatorId(),
                existingTicket.creationDate()
        ))
        .map(ticketRepository::save)
        .orElseThrow(() -> new NoSuchElementException(EXCEPTION_MESSAGE_PART1 + ticketId + EXCEPTION_MESSAGE_PART2));
```

Vlt hast du es noch aus dem React-Vortrag im Kopf: `Map` erstellt in Lambdas neue Kollektionen

Reply:
Falls du mehr dazu erfahren möchtest, kannst du ChatGPT ggf mit folgendem Propt fragen:

"Is it possible to make this code more elegant by using the optional syntax sugar?
<dein code hier>"